tests/app/dao/test_service_letter_contact_dao.py

```diff
@@ -0,0 +1,202 @@
+    create_letter_contact(service=service, contact_block='Edinburgh, ED1 1AA')
+    create_letter_contact(service=service, contact_block='Manchester, MA1 2BB', is_default=False)
+
+    with pytest.raises(expected_exception=InvalidRequest) as e:
+        create_or_update_letter_contact(service_id=service.id, contact_block='Swansea, SN1 3CC')
+    assert e.value.message == "Multiple letter contacts were found, this method should not be used."
+
+
+def test_add_letter_contact_for_service_creates_additional_letter_contact_for_service(notify_db_session):
+    service = create_service()
+
+    create_letter_contact(service=service, contact_block='Edinburgh, ED1 1AA')
+    add_letter_contact_for_service(service_id=service.id, contact_block='Swansea, SN1 3CC', is_default=False)
+
+    results = dao_get_letter_contacts_by_service_id(service_id=service.id)
+
+    assert len(results) == 2
+
+    assert results[0].contact_block == 'Edinburgh, ED1 1AA'
+    assert results[0].is_default
+
+    assert results[1].contact_block == 'Swansea, SN1 3CC'
+    assert not results[1].is_default
+
+
+def test_add_another_letter_contact_as_default_overrides_existing(notify_db_session):
+    service = create_service()
+
+    create_letter_contact(service=service, contact_block='Edinburgh, ED1 1AA')
+    add_letter_contact_for_service(service_id=service.id, contact_block='Swansea, SN1 3CC', is_default=True)
+
+    results = dao_get_letter_contacts_by_service_id(service_id=service.id)
+
+    assert len(results) == 2
+
+    assert results[0].contact_block == 'Edinburgh, ED1 1AA'
+    assert not results[0].is_default
+
+    assert results[1].contact_block == 'Swansea, SN1 3CC'
+    assert results[1].is_default
+
+
+def test_add_letter_contact_does_not_override_default(notify_db_session):
+    service = create_service()
+
+    add_letter_contact_for_service(service_id=service.id, contact_block='Edinburgh, ED1 1AA', is_default=True)
+    add_letter_contact_for_service(service_id=service.id, contact_block='Swansea, SN1 3CC', is_default=False)
+
+    results = dao_get_letter_contacts_by_service_id(service_id=service.id)
+
+    assert len(results) == 2
+
+    assert results[0].contact_block == 'Edinburgh, ED1 1AA'
+    assert results[0].is_default
+
+    assert results[1].contact_block == 'Swansea, SN1 3CC'
+    assert not results[1].is_default
+
+
+def test_add_letter_contact_with_no_default_raises_exception(notify_db_session):
+    service = create_service()
+    with pytest.raises(expected_exception=InvalidRequest):
+        add_letter_contact_for_service(
+            service_id=service.id,
+            contact_block='Swansea, SN1 3CC',
+            is_default=False
+        )
+
+
+def test_add_letter_contact_when_multiple_defaults_exist_raises_exception(notify_db_session):
+    service = create_service()
+    create_letter_contact(service=service, contact_block='Edinburgh, ED1 1AA')
+    create_letter_contact(service=service, contact_block='Aberdeen, AB12 23X')
```

Comment:
Would it be better to override the first as the default? I think this is the behaviour Becca adopted for multiple senders? Might be worth being consistent? (this might be consistent with somewhere else?

Reply:
I don't think it matters too much. Key thing is the purpose of the test is clear and the inherent code. I like to think this is pretty concise?